### ISSUE TITLE
Deja Dup fails doing a recovery test with an "unknown error"

### ISSUE THREAD

**Issue by NoTuxNoBux** [jstaf/onedriver#216]
Hello

This is perhaps a niche case, but I'm using OneDriver in combination with Déjà Dup (GNOME backups) to back-up files in an encryption fashion to OneDrive. The back-up itself works totally fine after the mount has been made, but every couple of back-ups, Déjà Dup wants to do a recovery test, which always fails with an "unknown error" on OneDriver.

This is not a large problem currently, since the actual back-up appears to work, but I just thought I'd mention it. Perhaps it is trying to use a feature that is not yet supported. It also prints no logs to the journal, so there is no immediate problem pointing to the cause.

**Comment by jstaf** [comment]
Do you see any errors in the onedriver logs? Even though Deja Dup might not log anything, onedriver will probably log whatever the problem was.

**Comment by NoTuxNoBux** [comment]
The newest version of Déjà Dup prints the following error:

```
Traceback (innermost last):
  File "/app/bin/duplicity", line 87, in <module>
    with_tempdir(main)
  File "/app/bin/duplicity", line 70, in with_tempdir
    fn()
  File "/app/lib/python3.9/site-packages/duplicity/dup_main.py", line 1568, in main
    do_backup(action)
  File "/app/lib/python3.9/site-packages/duplicity/dup_main.py", line 1651, in do_backup
    restore(col_stats)
  File "/app/lib/python3.9/site-packages/duplicity/dup_main.py", line 736, in restore
    restore_get_patched_rop_iter(col_stats)):
  File "/app/lib/python3.9/site-packages/duplicity/dup_main.py", line 788, in restore_get_patched_rop_iter
    file_names.append(backup_set.volume_name_dict[vol_num])
 KeyError: 1
```

OneDriver itself doesn't seem to print anything of interest relating to the error, just the standard `INF` and `DBG` output around fetching deltas and uploading bytes.

**Comment by NoTuxNoBux** [comment]
I think this may be fixed now with 0.13.0 or another recent release. I just had a recovery test again on 0.13.0 and it succeeded successfully.

**Comment by NoTuxNoBux** [comment]
Going to close this; I haven't seen this in a while.
